src/bvh/bvh_impl.rs

```diff
@@ -436,7 +437,7 @@ pub struct Bvh<T: Scalar + Copy, const D: usize> {
     pub nodes: Vec<BvhNode<T, D>>,
 }
 
-impl<T: Scalar + Copy, const D: usize> Bvh<T, D> {
+impl<T: Scalar + Copy  + Float, const D: usize> Bvh<T, D> {
```

Comment:
```suggestion
impl<T: Scalar + Copy, const D: usize> Bvh<T, D> {
```

Should not be bounded by Float (also better to use RealField), there are operations that can potentially work for integers

Reply:
RealField doesn't have infinity methods fyi
![image](https://user-images.githubusercontent.com/64865196/236619847-97d5e2c2-398c-4f89-a655-d84b1f1e5eb0.png)

Reply:
I could replace Infinity with minus one.